pg_ivm.c

```diff
@@ -326,10 +326,9 @@ CreateChangePreventTrigger(Oid matviewOid)
 Oid
 PgIvmImmvRelationId(void)
 {
-	if (!OidIsValid(pg_ivm_immv_id))
-		pg_ivm_immv_id = get_relname_relid("pg_ivm_immv", PG_CATALOG_NAMESPACE);
-
-	return pg_ivm_immv_id;
+	return RangeVarGetRelidExtended(
+		makeRangeVar("pg_catalog", "pg_ivm_immv", -1),
+		AccessShareLock, RVR_MISSING_OK, NULL, NULL);
 }
```

Comment:
I've understood that RangeVarGetRelid is required to avoid the ` could not open relation ` ERROR, but PgIvmImmvRelationId is called from another places in pg_ivm, so I would not like to acquire a lock in this function for avoiding unnecessary behaviour changes. Instead, I would like to call RangeVarGetRelid() directly in PgIvmObjectAccessHook. (I think it is better to use RangeVarGetRelid rather than RangeVarGetRelidExtended for making the function call simpler.)

Reply:
> I would not like to acquire a lock in this function for avoiding unnecessary behaviour changes.

After some considering, I came to think that this is not problem because the lock only conflicts with DROP EXTENSION. Moreover,  where PgIvmImmvRelationId is called, AccessShareLock or more stronger lock is acquired right after it, so taking a lock in this function would not cause additional conflicts including dead locks.